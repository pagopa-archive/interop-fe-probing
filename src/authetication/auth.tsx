import awsmobile from './aws-exports'
import { Auth, Amplify } from 'aws-amplify'

Amplify.configure(awsmobile)

async function login(username: string, password: string) {
  try {
    const user = await Auth.signIn(username, password)
    const token = user.signInUserSession.idToken.jwtToken
    const refreshToken = user.signInUserSession.refreshToken.token
    const accessToken = user.signInUserSession.accessToken.jwtToken
    sessionStorage.setItem('token', token)
    sessionStorage.setItem('refreshToken', refreshToken)
    sessionStorage.setItem('accessToken', accessToken)
    return user
  } catch (error) {
    console.log('error signing in', error)
  }
}

async function logout() {
  try {
    await Auth.signOut()
    sessionStorage.clear()
  } catch (error) {
    console.log('error signing out: ', error)
  }
}

export { login, logout }
