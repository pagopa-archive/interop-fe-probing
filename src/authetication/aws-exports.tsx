/* eslint-disable */
// WARNING: DO NOT EDIT. This file is automatically generated by AWS Amplify. It will be overwritten.

const awsmobile = {
  aws_project_region: import.meta.env.VITE_AWS_PROJECT_REGION,
  aws_cognito_identity_pool_id: null,
  aws_cognito_region: import.meta.env.VITE_AWS_COGNITO_REGION,
  aws_user_pools_id: import.meta.env.VITE_AWS_USER_POOL_ID,
  aws_user_pools_web_client_id: import.meta.env.VITE_AWS_USER_POOLS_WEB_CLIENT_ID,
  oauth: {},
  aws_cognito_username_attributes: ['EMAIL'],
  aws_cognito_social_providers: [],
  aws_cognito_signup_attributes: ['EMAIL'],
  aws_cognito_mfa_configuration: 'OFF',
  aws_cognito_mfa_types: [],
  aws_cognito_password_protection_settings: {
    passwordPolicyMinLength: 8,
    passwordPolicyCharacters: [],
  },
  aws_cognito_verification_mechanisms: ['EMAIL'],
}

export default awsmobile
