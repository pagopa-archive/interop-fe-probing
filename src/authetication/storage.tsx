const setStorage = (name: string, value: string) => {
  return new Promise((resolve, reject) => {
    sessionStorage.setItem(name, value);
    return resolve("OK");
  });
};

const deleteStorage = (name: string) => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.removeItem(name));
  });
};

const resetStorage = () => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.clear());
  });
};

const getStorage = (item: string) => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.getItem(item));
  });
};

export { setStorage, deleteStorage, resetStorage, getStorage };
