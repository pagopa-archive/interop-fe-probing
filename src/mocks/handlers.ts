import { rest } from 'msw'

export const handlers = [
  rest.get('http://localhost:3000/eservices', (req, res, ctx) => {
    const mockResponse = {
      content: [
        {
          id: '2',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Torino',
          state: 'OFFLINE',
          versionNumber: 8,
        },
        {
          id: '1',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Milano',
          state: 'ONLINE',
          versionNumber: 7,
        },
        {
          id: '3',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Torino',
          state: 'ONLINE',
          versionNumber: 7,
        },
        {
          id: '10',
          eserviceName: 'Erogazione borse di studio',
          producerName:
            'A.Di.S.U.R.C. Azienda per Il Diritto allo Studio Universitario della Regione Campania',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '11',
          eserviceName: 'Funzionamento pannelli solari',
          producerName: 'Comune di Brindisi',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '4',
          eserviceName:
            'Posizione centraline ambientali con un nome davvero molto molto molto lungo',
          producerName: 'Comune di Pescara',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '5',
          eserviceName: 'Postazioni e-bike',
          producerName: 'Comune di Roma',
          state: 'OFFLINE',
          versionNumber: 1,
        },
        {
          id: '6',
          eserviceName: 'Stato colonnine di ricarica auto elettriche',
          producerName: 'Comune di Bari',
          state: 'OFFLINE',
          versionNumber: 1,
        },
        {
          id: '7',
          eserviceName: 'Stato lavori Olimpiadi 2026',
          producerName: 'Comune di Milano',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '8',
          eserviceName: 'Stato opere PGT 2030',
          producerName: 'Comune di Trento',
          state: 'n/d',
          versionNumber: 1,
        },
      ],
      offset: 0,
      limit: 10,
      totalElements: 11,
    }

    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/producers', (req, res, ctx) => {
    const mockResponse = [
      {
        label: 'producer2',
        value: 'producer2',
      },
    ]
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/eservices/mainData/', (req, res, ctx) => {
    const mockResponse = {
      eserviceName: 'Probing test 3',
      producerName: 'Comune di Milano',
      versionNumber: '7',
      pollingFrequency: 5,
    }
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/eservices/probingData/', (req, res, ctx) => {
    const mockResponse = {
      probingEnabled: true,
      state: 'offline',
      responseReceived: '2023-04-27T14:55:25Z',
      eserviceActive: false,
    }
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/telemetryData/eservices/', (req, res, ctx) => {
    const mockResponse = {
      values: [
        {
          responseTime: 640,
          status: 'OK',
          time: '2023-05-16T13:17:01.579Z',
        },
        {
          responseTime: 98,
          status: 'OK',
          time: '2023-05-16T13:33:07.271Z',
        },
        {
          responseTime: 6,
          status: 'OK',
          time: '2023-05-16T13:39:01.371Z',
        },
      ],
      percentages: [
        {
          value: 94.28572,
          status: 'OK',
        },
        {
          value: 5.714286,
          status: 'KO',
        },
      ],
    }
    return res(ctx.json(mockResponse))
  }),
]
