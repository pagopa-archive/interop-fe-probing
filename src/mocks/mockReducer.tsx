import { BrowserRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

const queryClient = new QueryClient()

function reducer(ui: any, { preloadedState, store, ...renderOptions }: any = {}) {
  function Wrapper({ children }: any) {
    return (
      <Router>
        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
      </Router>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

export { reducer }
