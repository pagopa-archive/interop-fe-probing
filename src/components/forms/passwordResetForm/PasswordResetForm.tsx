import { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { passwordReset } from '../../../authetication/auth'
import { Form } from '../../form/Form'
import stores from '../../../store/Store'
import { Spinner } from '../../spinner/Spinner'
import queryString from 'query-string'

interface IProps {
  setResetSuccess: Function
}

/**
 * default values of the form fields
 */
const defaultFormValues: { [key: string]: string } = {
  password: '',
  confirmPassword: '',
}

const PASSWORD_REGEX =
  /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!.* )[\w!@#$%^&()+."\-\?{}`~[\]*=|;:'<>,\/\\]{12,}$/

/**
 * Generating the password reset form
 * @component
 */
export const PasswordResetForm: React.FC<IProps> = ({ setResetSuccess }) => {
  const [spinner, setSpinner] = useState(false)

  const { t } = useTranslation(['passwordResetPage'])

  const [updateSnackbar] = stores.useSnackbarStore((state) => [state.updateSnackbar])

  /**
   * properties of the form fields
   */
  const fieldsProperties = [
    {
      name: 'password',
      label: t('newPassword', { ns: 'passwordResetPage' }),
      type: 'password',
      rules: {
        minLength: {
          value: 12,
          message: 'passwordTooShortError',
        },
        pattern: {
          value: PASSWORD_REGEX,
          message: 'passwordRegexError',
        },
      },
      width: '500px',
    },
    {
      name: 'confirmPassword',
      label: t('confirmNewPassword', { ns: 'passwordResetPage' }),
      type: 'password',
      rules: {
        minLength: {
          value: 12,
          message: 'passwordTooShortError',
        },
        pattern: {
          value: PASSWORD_REGEX,
          message: 'passwordRegexError',
        },
        validate: {
          validateEquality: (_: string, values: { [key: string]: string }) =>
            values.password === _ || 'passwordEqualityError',
        },
      },
      width: '500px',
    },
  ]

  const onSubmit = async (data: { [key: string]: string }) => {
    const code = queryString.parse(location.hash).code?.toString() || ''
    const email = queryString.parse(location.hash).username?.toString().replace('%', '@') || ''
    setSpinner(true)
    passwordReset(email, code, data.password)
      .then(() => {
        setSpinner(false)
        setResetSuccess(true)
      })
      .catch((error) => {
        setSpinner(false)
        const message = t('resetError', { ns: 'passwordResetPage' })
        updateSnackbar(true, message, 'error')
      })
  }

  return (
    <>
      <Form
        fields={fieldsProperties}
        submitButton={t('submit', { ns: 'passwordResetPage' })}
        submitFunction={onSubmit}
        defaultValues={defaultFormValues}
        buttonDisable={'never'}
        validationMode={'onSubmit'}
        t={t}
      />
      <Spinner
        open={spinner}
        setSpinner={setSpinner}
        message={t('spinnerMessage', { ns: 'passwordResetPage' })}
      />
    </>
  )
}
