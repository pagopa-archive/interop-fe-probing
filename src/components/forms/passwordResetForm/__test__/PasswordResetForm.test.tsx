import { PasswordResetForm } from '../PasswordResetForm'
import { expect, test, vi } from 'vitest'
import { renderWithRouterAndQuery } from '../../../../mocks/mockWrapper'
import { cleanup, screen, logRoles } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import * as auth from '../../../../authetication/auth'

describe('PasswordResetForm', () => {
  afterEach(cleanup)

  test('render component', () => {
    const { container } = renderWithRouterAndQuery(<PasswordResetForm setResetSuccess={vi.fn()} />)
    expect(container).toMatchSnapshot()
  })

  test('fill field and click submit simulating success', async () => {
    const spy = vi.spyOn(auth, 'passwordReset')
    spy.mockImplementation(() => Promise.resolve())

    renderWithRouterAndQuery(<PasswordResetForm setResetSuccess={vi.fn()} />)
    const submitButton = screen.getByRole('button')
    const passwordField = document.getElementById('password') as HTMLInputElement
    const confirmPasswordField = document.getElementById('confirmPassword') as HTMLInputElement
    const user = userEvent.setup()
    await user.type(passwordField, 'Test@password123.')
    await user.type(confirmPasswordField, 'Test@password123.')
    await user.click(submitButton)
    expect(spy).toBeCalled()
  })

  test('fill field and click submit simulating error', async () => {
    vi.spyOn(auth, 'passwordReset').mockImplementation(async () => {
      throw new Error('UNKNOWN ERROR')
    })

    renderWithRouterAndQuery(<PasswordResetForm setResetSuccess={vi.fn()} />)
    const submitButton = screen.getByRole('button')
    const passwordField = document.getElementById('password') as HTMLInputElement
    const confirmPasswordField = document.getElementById('confirmPassword') as HTMLInputElement
    const user = userEvent.setup()
    await user.type(passwordField, 'Test@password123.')
    await user.type(confirmPasswordField, 'Test@password123.')

    try {
      await user.click(submitButton)
    } catch (error: unknown) {
      if (error instanceof Error) {
        expect(error.message).toBe('UNKNOWN ERROR')
      }
    }
  })
})
