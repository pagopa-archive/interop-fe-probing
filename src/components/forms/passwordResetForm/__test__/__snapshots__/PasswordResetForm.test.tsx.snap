// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PasswordResetForm > render component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation16 MuiCard-root css-9cvwhb-MuiPaper-root-MuiCard-root"
  >
    <form>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-ljy4ha-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item css-1qy0967-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-p21bt9-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                newPassword
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      newPassword
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-p21bt9-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="confirmPassword"
                id="confirmPassword-label"
              >
                confirmNewPassword
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="confirmPassword"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      confirmNewPassword
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-131xi47-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-w4y956-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-jb1ygz-MuiTypography-root"
            >
               
              submit
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
